### DISCUSSION TITLE
Pinia performance ?

### DISCUSSION THREAD

**Discussion by thealirexa** [vuejs/pinia#411] [Help and Questions] [answered]
Hi guys,
I have a big project that has a lot of stores (for example 100 store), if i call a large number of these stores in one component, after the component is destroyed, are the stores still in RAM?
can leak memory problem occur? or is the app screens lagging?

**Comment by posva** [comment, marked as answer]
only a partial store is always kept. Namely, the state is preserved (that's the point of having a store). You are free to free the data at any moment to reduce memory consumption by resetting a big array or setting a value to `null`.

Lags should not happen. One can also use `markRaw()` for data that doesn't change
